gateway/manager/targetcluster/registry.go

```diff
@@ -218,6 +228,106 @@ func (cr *ClusterRegistry) handleCORS(w http.ResponseWriter, r *http.Request) bo
 	return false
 }
 
+func (cr *ClusterRegistry) validateToken(token string, cluster *TargetCluster) (bool, error) {
```

Comment:
This function is big and duplicates what we have in roundtripper.
I am trying to simplify it by using rountripper in its request.